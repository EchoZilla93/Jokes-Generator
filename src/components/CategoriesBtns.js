import React,{useState} from 'react';
import {useDispatch} from 'react-redux';

import {getAnimalJoke,getCareerJoke,getCelebrityJoke,getDevJoke} from '../actions/index';



const CategoriesBtns = () => {
    const dispatch = useDispatch();
    const [btnClick,setBtnClick] = useState(false);

   //Btn Controllers:
   const onAnimalClick = () => dispatch(getAnimalJoke());
   const onCareerClick = () => dispatch(getCareerJoke());
   const onCelebrityClick = () => dispatch(getCelebrityJoke());
   const onDevClick = () => dispatch(getDevJoke());

    return(
        <span className="btnContainer">
            <button 
            className={`btnContainer__btn animal  ${btnClick? 'active': 'nonActive'}`} 
            onClick={onAnimalClick}>animal</button>
            <button 
            className={`btnContainer__btn career ${btnClick? 'active': 'nonActive'}`} 
            onClick={onCareerClick}>career</button>
            <button 
            className={`btnContainer__btn ${btnClick? 'active': 'nonActive'}`} 
            onClick={onCelebrityClick}>celebrity</button>
            <button 
            className={`btnContainer__btn ${btnClick? 'active': 'nonActive'}`} 
            onClick={onDevClick}>dev</button>
        </span>
    );
};

export default CategoriesBtns;