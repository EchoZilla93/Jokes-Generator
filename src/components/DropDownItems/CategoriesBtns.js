import React from 'react';
import {useDispatch} from 'react-redux';

import {setCategory} from '../../actions/index';
import './dropDownStyles.scss';



const CategoriesBtns = () => {
    const dispatch = useDispatch();
    const setJokeCategory = (e , value) => {
        e.preventDefault();
        dispatch(setCategory(value));
    }

    return(
        <span className="btnContainer">
            <button 
            className={'btnContainer__btn animal '} 
            onClick={e => setJokeCategory(e, 'animal')}>animal</button>
            <button 
            className={`btnContainer__btn career`} 
            onClick={e => setJokeCategory(e, 'career')}>career</button>
            <button 
            className={`btnContainer__btn`} 
            onClick={e => setJokeCategory(e, 'celebrity')}>celebrity</button>
            <button 
            className={`btnContainer__btn`} 
            onClick={e => setJokeCategory(e, 'dev')}>dev</button>
        </span>
    );
};

export default CategoriesBtns;