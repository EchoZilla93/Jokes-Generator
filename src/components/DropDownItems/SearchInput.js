import React from 'react';
import {useDispatch} from 'react-redux';

import {setCategory} from '../../actions/index';
import './dropDownStyles.scss';

export const SearchInput = () => {
    const dispatch = useDispatch();

    const setSpecificCategory = value =>{
        dispatch(setCategory(value));
    };

    return (
        <div className="searchInput">
            <input 
                className="searchInput__field" 
                type="text" 
                placeholder="Free text search..." 
                onChange={e => setSpecificCategory(e.target.value)}/>
        </div>
    )
}

export default SearchInput;