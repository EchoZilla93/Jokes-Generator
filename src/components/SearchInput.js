import React from 'react';

const SearchInput = () => {
    return (
        <div className="searchInput">
            <input className="searchInput__field" type="text" placeholder="Free text search..."/>
        </div>
    )
}

export default SearchInput;