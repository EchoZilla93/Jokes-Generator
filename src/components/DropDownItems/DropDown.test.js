import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { CategoriesBtns } from './CategoriesBtns';
import { SearchInput } from './SearchInput';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('DropDown items', () => {
    const initialState = {
        jokeCategoryToSearch: 'animal'
    };
    const mockStore = configureStore();
    let store, wrapper;
    it('buttons renders expected snapshot', () => {
        store = mockStore(initialState);
        wrapper = shallow(
            <Provider store={store}>
                <CategoriesBtns />
            </Provider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('search input renders expected snapsot', () => {
        store = mockStore(initialState);
        wrapper = shallow(
            <Provider store={store}>
                <SearchInput />
            </Provider>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});