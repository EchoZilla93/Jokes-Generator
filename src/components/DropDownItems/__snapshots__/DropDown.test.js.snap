// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown items buttons renders expected snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <CategoriesBtns />
</ContextProvider>
`;

exports[`DropDown items search input renders expected snapsot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <SearchInput />
</ContextProvider>
`;
